app/server.go

```diff
@@ -291,20 +293,29 @@ func askReachableAddress(port string) network.Address {
 // rules etc).
 // In case anything goes wrong, an error is returned.
 func tryConnect(ip, binding network.Address) error {
-
 	stopCh := make(chan bool, 1)
+	listening := make(chan bool)
 	// let's bind
 	go func() {
 		ln, err := net.Listen("tcp", binding.NetworkAddress())
 		if err != nil {
 			log.Error("Trouble with binding to the address:", err)
 			return
 		}
+		listening <- true
 		con, _ := ln.Accept()
 		<-stopCh
 		con.Close()
 	}()
 	defer func() { stopCh <- true }()
+	<-listening
+	conn, err := net.Dial("tcp", ip.NetworkAddress())
+	if err != nil {
+		log.Error("Could not connect itself to public address:", err)
```

Comment:
I think this should `Panic` and not just `Error`. It's an error that needs to be corrected.
Add a hint for sysadmins:
"This is most probably an error in your system-setup. Please make sure this conode can connect to #{address}"